R/rank.R

```diff
@@ -36,9 +33,13 @@
 #'   non-duplicate value will have a rank incremented by `1`, effectively
 #'   removing any gaps in the ranking.
 #'
-#' @param na_propagate A single logical specifying whether or not missing
-#'   values should be propagated. If `TRUE`, all missing values are given
-#'   the rank `NA`.
+#' @param incomplete Treatment of [incomplete][vec_detect_complete] values.
+#'
+#'   - `"rank"`: Give incomplete values a numeric rank, which is affected
+#'   by `na_value` and `nan_distinct`.
+#'
+#'   - `"na"`: Give incomplete values a rank of `NA`, regardless of the value
```

Comment:
I don't understand this part. Won't incomplete rows be affected by `na_value` if they are treated as `NA`?

Reply:
They aren't treated like `NA`. We don't rank them at all, instead giving them a value of `NA` in the result.

Reply:
We actually rank them when `incomplete = "rank"` is set. Which is when `na_value` and `nan_distinct` comes into play, because that determines how missing values rank vs non-missing values, and vs `NaN` values